src/utils/tree.ts

```diff
@@ -0,0 +1,72 @@
+import { type ColorName, getColor } from "./color";
+
+export type TreeItemObject = {
+  /**
+   * Text of the item
+   */
+  text: string;
+  /**
+   * Color of the item
+   */
+  color?: ColorName;
+};
+export type TreeItem = string | TreeItemObject;
+
+export type TreeOptions = {
+  /**
+   * Color of the tree
+   */
+  color?: ColorName;
+  /**
+   * Prefix of the tree
+   *
+   * @default "  "
+   */
+  prefix?: string;
+};
+
+export function tree(items: TreeItem[], options?: TreeOptions): string {
+  options = {
+    prefix: "  ",
+    ...options,
+  };
+
+  const tree = formatTree(items, options);
+  if (options && options.color) {
+    const colorize = getColor(options.color);
+    return colorize(tree);
+  }
+
+  return tree;
+}
+
+function formatTree(items: TreeItem[], options?: TreeOptions): string {
+  let logs = "";
+
+  const total = items.length - 1;
+  for (const item of items) {
+    const isLast = items.indexOf(item) === total;
```

Comment:
Performance: Usually `for of` is best syntax but here if we use a `for (let i` we can simply check last index (index of makes `O(n^2)` as it has to search item in whole array each time)

Reply:
Yep, totally. I will change this.